### ISSUE TITLE
Project Organization/Architecture (Generic) 

### ISSUE THREAD

**Issue by bitrecluse** [dawnbomb/Crystal-Editor#1]
**Issue:**
This project seems to lack a clear cut direction in software architecture which may make it hard for new comers to the repository to quickly come up to speed.

**Suggestion:**
A quick fix may be to spend some time thinking of folder naming. Here in the root code bas there are folders such as:

"Z Garbage"
"Properties"
"Menus"

Just to cherry pick a few. Something like "Garbage" has no real definition to someone outside of the code base. Why is it garbage? Why is it here? 
Something like "Properties" is very generic. properties for what? is this UI? is this a Model? is this DB properties?
Something like "Menus" is obviously UI related, but why is front end mixed with back-end? Why not UI>Menus | UI>Properties?

Obviously a person can do their due diligence and learn the code-base over time, but as a somewhat open-source ish repository it needs to meet generally accepted naming patterns and software architecture.

**Comment by dawnbomb** [comment]
again i am new to C# and coding, this is my first project.

I don't understand basically every part of this issues.
i don't understand what you mean by a clear cut direction.
i don't understand what you mean by software architecture.
However, ... if by those you mean "I don't understand the folder structure", yes i know. thats why i did a little writeup on the site explaining stuff, and i'm trying to improve it a little, but i'm a bit more focused on progress over organization right now.

Z Garbage is a bunch of unimportant garbage i'd rather not even be in the project but i can't really remove.
Properties....i don't believe i made that, i think it's part of a default new project. I don't know what repercussions there are of renaming, moving, or deleting default project assets.
Menus is part of the Menus rework i'm working on, it's not done yet. (Most would understand because i am directly telling anyone this, and it's also literally the first point on my goal bulletin on the google sheet you don't want to use)

i know garbage has no definition to people outside, they would have to ask a question, this is unavoidable. I suppose i can put an entire paragraph for a folder name giving a detailed explanation, but thats kind of stupid and noone wants folder names like that.

(Properties is generic..) complain to visual studio then, i didn't make it.
(Menus is mixing front and back end..) I was just attempting my best to put most of the stuff related to that menu, inside a menu folder, so people aren't looking around. I don't understand what you mean by UI > Menus, my best guess is you want all UI windows together, and then i think it would be hard to follow anything. Currently, i am sorting things (attemptedly) by what window they are, and what their related to. Window 1, the game library, gets its own folder. Window 2, the workshop, gets its own folder. Window 3, the wiki, gets its own folder.  Menus, being a user control that is basically its own window, gets its own folder.

**Comment by bitrecluse** [comment]
I understand you did some things out away from your source code, but those things are useless when a developer is inside of your source code.
Asking a developer to tab back and forth will almost certainly mean they get frustrated very quickly with the codebase.
I know you cant really change your code on Github, but you can still see all of it, review it, talk about it, etc.

> (Properties is generic..) complain to visual studio then, i didn't make it.

This just is not an acceptable attitude for a developer. a developer is responsible for the smallest bit (whether its a 1 or a 0) of their project. If a tool you used created some ginormous auto-generate monolith of a file, well then you are now responsible for that.

**Comment by dawnbomb** [comment]
Those things are useless to you because you intentionally refuse to look at them or speak in a normal way and left discord. those things are not useless when "a developer" is inside, only when it's someone being intentionally as obtuse, uncooperative, and unwilling to speak to other people in normal conversations as you are. *normal* people don't intentionally avoid any kind of goal and only complain. 

(This just is not an acceptable attitude for a developer)... WHAT part is so unacceptable you moron. The "I am new and don't know what will happen" or the "its default in visual studio / i didn't make it" part. If you don't like microsoft making it, complain to them, not to me. But weither it's because VS made it by default, or because i'm new, my answer is the same. 

Whats not acceptable is how you treat me, and why i'm being so openly mean to you now. Who the fuck hears someone is new and tells them it's "unacceptable", mother fucker, this is a voluntary assistance github. Where do you get off coming into my project, and telling me that MY ATTTUDE is unacceptable? Bitch, you blatently ignore me for days, gaslight the fuck out of me on literally every topic i tried discussing with you, brag about your fucking house, made me argue for hours over why the programs exe has folders next to it (like almost literally every program ever made), but when someone new to coding says he didn't make the properties folder and don't know what will happen, that "unacceptable"? fuck off.

**Comment by bitrecluse** [comment]
> (This just is not an acceptable attitude for a developer)... WHAT part is so unacceptable you moron. The "I am new and don't know what will happen" or the "its default in visual studio / i didn't make it" part.

This is my point. How exactly can you learn if you never ever accept you might have made a mistake.

**Comment by bitrecluse** [comment]
The answer is that you cannot and now I think that I will indeed leave

**Comment by dawnbomb** [comment]
You have to be mentally retarded. That "i" made a mistake... i said the properties folder is default, and i don't know what will happen if i touch it. There is, objectively, no "mistake" anywhere there. I am not "mistaken" in that i don't know what will happen, nor am i "mistaken" that it is created by default. This is you just gaslighting me more.
